File: VB/AccessActionControl/AccessActionControl.Module/MyDomainObject.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel

Imports DevExpress.Xpo

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace AccessActionControl.Module
	<DefaultClassOptions> _
	Public Class MyDomainObject
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		Public Property CreatedOn() As DateTime
			Get
				Return GetPropertyValue(Of DateTime)("CreatedOn")
			End Get
			Set(ByVal value As DateTime)
				SetPropertyValue("CreatedOn", value)
			End Set
		End Property

		Public Property Title() As String
			Get
				Return GetPropertyValue(Of String)("Title")
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Title", value)
			End Set
		End Property
	End Class
End Namespace

File: CS/AccessActionControl.Module/MyDomainObject.cs
using System;
using System.ComponentModel;

using DevExpress.Xpo;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace AccessActionControl.Module {
    [DefaultClassOptions]
    public class MyDomainObject : BaseObject {
        public MyDomainObject(Session session) : base(session) { }

        public DateTime CreatedOn {
            get { return GetPropertyValue<DateTime>("CreatedOn"); }
            set { SetPropertyValue("CreatedOn", value); }
        }

        public string Title {
            get { return GetPropertyValue<string>("Title"); }
            set { SetPropertyValue("Title", value); }
        }
    }
}
